src/components/Engines/Engines.tsx

```diff
@@ -1,73 +1,51 @@
-import React, { Component, ReactElement } from 'react';
+import React, { useContext } from 'react';
 import Grid from '@material-ui/core/Grid';
 import ListItemText from '@material-ui/core/ListItemText';
 
-import { VersionPageConsumerProps, DetailContextConsumer } from '../../pages/Version';
+import { DetailContext } from '../../pages/Version';
 import Avatar from '../../muiComponents/Avatar';
 import List from '../../muiComponents/List';
 import npm from '../Install/img/npm.svg';
 
 import { StyledText, EngineListItem } from './styles';
-// @ts-ignore
 import node from './img/node.png';
 
-const ICONS = {
-  'node-JS': <Avatar src={node} />,
-  'NPM-version': <Avatar src={npm} />,
-};
-
-class Engine extends Component {
-  public render(): ReactElement<HTMLElement> {
-    return (
-      <DetailContextConsumer>
-        {context => {
-          return this.renderEngine(context as VersionPageConsumerProps);
-        }}
-      </DetailContextConsumer>
-    );
-  }
-
-  private renderEngine = ({ packageMeta }): ReactElement<HTMLElement> | null => {
-    const { engines } = packageMeta.latest;
-    if (!engines) {
-      return null;
-    }
+const Engine: React.FC = () => {
+  const { packageMeta } = useContext(DetailContext);
 
-    const engineDict = {
-      'node-JS': engines.node,
-      'NPM-version': engines.npm,
-    };
+  const engines = packageMeta && packageMeta.latest && packageMeta.latest.engines;
 
-    const accumulator: React.ReactNode[] = [];
-    const items = Object.keys(engineDict).reduce((markup, text, key) => {
-      const heading = engineDict[text];
-      if (heading) {
-        markup.push(
-          <Grid item={true} key={key} xs={6}>
-            {this.renderListItems(heading, text)}
-          </Grid>
-        );
-      }
-      return markup;
-    }, accumulator);
-
-    if (items.length < 1) {
-      return null;
-    }
-
-    return <Grid container={true}>{items}</Grid>;
-  };
+  if (!engines || (!engines.node && !engines.npm)) {
+    return null;
+  }
 
-  private renderListItems = (heading: string, text: string) => {
-    return (
-      <List subheader={<StyledText variant={'subtitle1'}>{text.split('-').join(' ')}</StyledText>}>
-        <EngineListItem button={true}>
-          {ICONS[text]}
-          <ListItemText primary={heading} />
-        </EngineListItem>
-      </List>
-    );
-  };
-}
+  /* eslint-disable react/jsx-max-depth */
```

Comment:
Disabled this because I think splitting the render function would harm readability.

Happy to change if anyone disagrees.

Reply:
Agree, rules has exceptions, this one is one of them.